### ISSUE TITLE
Scroll Reverser prevents "three finger tap" dictionary gesture working

### ISSUE THREAD

**Issue by invariant** [pilotmoon/Scroll-Reverser#1]
https://secure.helpscout.net/conversation/3174023/11/

The main problem is three finger tapping isn't working!

could you fix this problem? i will wait for your answer ;)

Thank you.

(and many more like it)

**Comment by allen-zh** [comment]
+1 hope it fixed

**Comment by pkamb** [comment]
I hit this same Three Finger Tap issue in PowerKey; I should have made the connection earlier.

Fix is to add additional event types to your `CGEventTap`'s the `CGEventMask` beyond the main event types you are listening for. 

In my case I ended up ignoring the below `NSKeyUp` etc. event types, but adding all others to the mask.

https://github.com/pkamb/PowerKey/blob/master/PowerKey/PKPowerKeyEventListener.m#L40

```
    CGEventMask eventTypeMask = NSSystemDefined;

    /*
     The power key sends events of type NSSystemDefined.
     We'd idealy monitor *only* NSSystemDefined events.
     But there are various bugs with certain other applications if we do.
     Therefore, we need to grab other events as well.
    */
    
    for (NSEventType type = NSLeftMouseDown; type < NSEventTypeGesture; ++type) {
        switch (type) {
            case NSKeyDown:
            case NSKeyUp:
            case NSEventTypeRotate:
            case NSEventTypeBeginGesture:
            case NSEventTypeEndGesture:
                break;
            default:
                eventTypeMask |= NSEventMaskFromType(type);
        }
    }
```

https://github.com/pkamb/PowerKey/pull/5
https://github.com/pkamb/PowerKey/commit/e152ac1e34d0d6697bb835c08f53e4c5ef222ec0

**Comment by invariant** [comment]
Thanks for this @pkamb. Very interesting. Unfortunately NSEventMaskGesture is one of the events that is *do* need to track. I've experimented quickly with adding in more types and nothing so far. Good to hear I am not alone on this.

**Comment by invariant** [comment]
Some more info about this problem for people who may find this page:

* If you run BetterTouchTool all the same time as Scroll Reverser, the problem goes away. Very mysterious!

* ctrl-cmd-D also activate the dictionary lookup, if you didn't know

if anyone has any more ideas about how to fix this, please do share.

**Comment by invariant** [comment]
Thought I had it, but 28a4d56 was a false alarm.

**Comment by katsos** [comment]
_Just to inform..._ 
I had the natural scroll off and I reversed the touchpad vertical and horizontal.
That gave me the **problem with the 3-finger gesture**.

I **solved** it, **enabling natural scroll and reversing mouse instead of trackpad**.

**Comment by invariant** [comment]
Further note. This seems to be a non-issue on (OS X 10.11 El Cap) and above.

**Comment by katsos** [comment]
@invariant I encountered it and I am on a Mac with OS X 10.11.6

**Comment by invariant** [comment]
@katsos Was this a one off or are you able to reliably demonstrates Scroll Reverser interfering with 3-finger tap in 10.11?

Anyone else?

It seems to me Apple fixed it.

**Comment by katsos** [comment]
@invariant not anymore. just checked it. 

seems fixed.

**Comment by invariant** [comment]
Closing as this problem appears to be fixed on macOS 10.11 and 10.2.
